### ISSUE TITLE
Using glimmer components gives errors with ember-cli-typescript installed

### ISSUE THREAD

**Issue by fpauser** [emberjs/ember.js#18443]
Using `@glimmer/component` gives errors whith `ember-cli-typescript` installed:

Works `(app/components/ui-button.js)`:
```js
import Component from '@ember/component';

export default class UiButtonComponent extends Component {

}
```

Errors `(app/components/ui-button.js)`:
```ts
import Component from '@glimmer/component';

export default class UiButtonComponent extends Component {

}
```

```
$ ember s
node_modules/@glimmer/interfaces/dist/types/lib/dom/bounds.d.ts:1:43 - error TS2307: Cannot find module '@simple-dom/interface'.

1 import { SimpleElement, SimpleNode } from '@simple-dom/interface';
                                            ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/runtime/dist/types/lib/vm/update.d.ts:4:28 - error TS2307: Cannot find module '@simple-dom/interface'.

4 import { SimpleNode } from '@simple-dom/interface';
                             ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/runtime/dist/types/lib/vm/update.d.ts:46:29 - error TS2307: Cannot find module '@simple-dom/interface'.

46     parentElement(): import("@simple-dom/interface").SimpleElement;
                               ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/runtime/dist/types/lib/vm/attributes/index.d.ts:2:46 - error TS2307: Cannot find module '@simple-dom/interface'.

2 import { AttrNamespace, SimpleElement } from '@simple-dom/interface';
                                               ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/runtime/dist/types/lib/vm/attributes/dynamic.d.ts:2:46 - error TS2307: Cannot find module '@simple-dom/interface'.

2 import { AttrNamespace, SimpleElement } from '@simple-dom/interface';
                                               ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/runtime/dist/types/lib/bounds.d.ts:3:43 - error TS2307: Cannot find module '@simple-dom/interface'.

3 import { SimpleElement, SimpleNode } from '@simple-dom/interface';
                                            ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/runtime/dist/types/lib/component/curried-component.d.ts:6:87 - error TS2677: A type predicate's type must be assignable to its parameter's type.
  Type 'CurriedComponentDefinition' is not assignable to type 'Maybe<Dict<unknown>>'.
    Type 'CurriedComponentDefinition' is not assignable to type 'Dict<unknown>'.
      Index signature is missing in type 'CurriedComponentDefinition'.

6 export declare function isComponentDefinition(definition: Maybe<Dict>): definition is CurriedComponentDefinition;
                                                                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/runtime/dist/types/lib/dom/operations.d.ts:1:86 - error TS2307: Cannot find module '@simple-dom/interface'.

1 import { SimpleElement, SimpleDocument, SimpleNode, SimpleText, SimpleComment } from '@simple-dom/interface';
                                                                                       ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/runtime/dist/types/lib/dom/helper.d.ts:3:103 - error TS2307: Cannot find module '@simple-dom/interface'.

3 import { AttrNamespace, ElementNamespace, Namespace, SimpleDocument, SimpleElement, SimpleNode } from '@simple-dom/interface';
                                                                                                        ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/runtime/dist/types/lib/dom/props.d.ts:1:31 - error TS2307: Cannot find module '@simple-dom/interface'.

1 import { SimpleElement } from '@simple-dom/interface';
                                ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/runtime/dist/types/lib/environment.d.ts:4:62 - error TS2307: Cannot find module '@simple-dom/interface'.

4 import { AttrNamespace, SimpleDocument, SimpleElement } from '@simple-dom/interface';
                                                               ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/runtime/dist/types/lib/vm/element-builder.d.ts:3:109 - error TS2307: Cannot find module '@simple-dom/interface'.

3 import { AttrNamespace, SimpleComment, SimpleDocumentFragment, SimpleElement, SimpleNode, SimpleText } from '@simple-dom/interface';
                                                                                                              ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/runtime/dist/types/lib/vm/rehydrate-builder.d.ts:3:85 - error TS2307: Cannot find module '@simple-dom/interface'.

3 import { AttrNamespace, SimpleComment, SimpleElement, SimpleNode, SimpleText } from '@simple-dom/interface';
                                                                                      ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/interfaces/dist/types/lib/dom/changes.d.ts:7:8 - error TS2307: Cannot find module '@simple-dom/interface'.

7 } from '@simple-dom/interface';
         ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/interfaces/dist/types/lib/runtime/modifier.d.ts:6:31 - error TS2307: Cannot find module '@simple-dom/interface'.

6 import { SimpleElement } from '@simple-dom/interface';
                                ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/interfaces/dist/types/lib/runtime/environment.d.ts:3:46 - error TS2307: Cannot find module '@simple-dom/interface'.

3 import { AttrNamespace, SimpleElement } from '@simple-dom/interface';
                                               ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/interfaces/dist/types/lib/components/component-manager.d.ts:2:31 - error TS2307: Cannot find module '@simple-dom/interface'.

2 import { SimpleElement } from '@simple-dom/interface';
                                ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/interfaces/dist/types/lib/runtime/render.d.ts:1:43 - error TS2307: Cannot find module '@simple-dom/interface'.

1 import { SimpleElement, SimpleNode } from '@simple-dom/interface';
                                            ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/interfaces/dist/types/lib/dom/attributes.d.ts:8:8 - error TS2307: Cannot find module '@simple-dom/interface'.

8 } from '@simple-dom/interface';
         ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/interfaces/dist/types/lib/dom/tree-construction.d.ts:7:8 - error TS2307: Cannot find module '@simple-dom/interface'.

7 } from '@simple-dom/interface';
         ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/util/dist/types/lib/destroy.d.ts:3:69 - error TS2677: A type predicate's type must be assignable to its parameter's type.
  Type 'SymbolDestroyable' is not assignable to type 'Maybe<Dict<unknown>>'.
    Type 'SymbolDestroyable' is not assignable to type 'Dict<unknown>'.
      Index signature is missing in type 'SymbolDestroyable'.

3 export declare function isDestroyable(value: Maybe<Dict>): value is SymbolDestroyable;
                                                                      ~~~~~~~~~~~~~~~~~

node_modules/@glimmer/util/dist/types/lib/dom.d.ts:1:31 - error TS2307: Cannot find module '@simple-dom/interface'.

1 import { SimpleElement } from '@simple-dom/interface';
                                ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@glimmer/util/dist/types/lib/is-serialization-first-node.d.ts:1:28 - error TS2307: Cannot find module '@simple-dom/interface'.

1 import { SimpleNode } from '@simple-dom/interface';
                             ~~~~~~~~~~~~~~~~~~~~~~~


Build successful (2396ms) – Serving on http://localhost:4200/
```

**Comment by lifeart** [comment]
@fpauser try to install `@simple-dom/interface`

**Comment by rwjblue** [comment]
I think this is largely fixed with `@glimmer/component@1.0.0-beta.1`

**Comment by fpauser** [comment]
Upgrade to `@glimmer/component@1.0.0-beta.1` helped - thanks @rwjblue!
